### DISCUSSION TITLE
Store the queue on disk, reload after crash

### DISCUSSION THREAD

**Discussion by 42degrees** [comfyanonymous/ComfyUI#1856] [Ideas]
I get video card crashes occasionally (usually about 1 per hour).  The annoying thing is that I lose the entire queue when this happens (besides having to reboot the computer).  I assume that the queue items are JSON blocks and further assume they are only stored in memory currently.  If either of these assumptions is wrong, I'd love to hear about that.  Otherwise I'll try to find some time figuring out where the queue is stored and trying to move it to disk.  This would also probably indicate a need for a configuration parameter to say where the queue is stored.  Also a button to restore the queue from the saved files.

Thoughts?

**Comment by TijuanaKez** [comment]
We really need this for things like FILM VFI that kills the whole ComfyUI process regularly.

**Reply by xx88xx** [reply]
@42degrees @TijuanaKez 
Did this ever make any traction? This feature would be immensely useful

**Reply by TijuanaKez** [reply]
Implementing it is a bit beyond my current skillset.
I wouldn't assume it wouldn't be that hard for people who are familiar with Comfy code to move the queue data so it can be persistent.

I'll often queue up a bunch of things at night with long render times like upscaled AnimateDIff or SVD hoping to see if changes are for better or worse etc. A lot of errors will just jump to the next queue item which is great.
Other things like Frame Interpolation seem to kill the whole process and the queue is lost which is extremely frustrating.

Really hoping someone else with familiarity can move the queue data to place that would persist after the process is killed.

**Reply by BrechtCorbeel** [reply]
would be really great. I might try, but I need to look into the whole code of the application and know where to look.

**Reply by BrechtCorbeel** [reply]
this seems like it might work about to try it:

https://github.com/atmaranto/ComfyUI-SaveAsScript

**Reply by TijuanaKez** [reply]
Also this.
https://github.com/Satellile/yara
but really this should all just happen automatically.

**Comment by QuietNoise** [comment]
Try this:
https://github.com/QuietNoise/ComfyUI-Queue-Manager

**Comment by CCpt5** [comment]
How is this STILL not a feature? @huchenlei / @comfyanonymous  ?

No idea how saving/restoring a queue can be that hard to implement. I often spend 25minutes+ queuing stuff to run in batch and often something will choke due to a memory spike (IE: I didn't enable VAE tiling). Then I'm stuck. Can't "restart" comfy w/o completely clearing everything I spent time on queuing.

Please find a way to make this happen (this is a 2023 thread).

**Reply by QuietNoise** [reply]
It's actually harder than you think. It's not easy to add this feature because of the way the queue system is currently implemented. 
I know it firsthand because my initial attempt was to hack the fork and give it a PR but I ended up changing too many things. Hence I decided to make it a full blown extension with additional features. Feel free to try it: https://github.com/comfyanonymous/ComfyUI/discussions/1856#discussioncomment-13569865

**Reply by CCpt5** [reply]
Will do - thanks for the heads up!

**Reply by BrechtCorbeel** [reply]
> How is this STILL not a feature? @huchenlei / @comfyanonymous ?
> 
> No idea how saving/restoring a queue can be that hard to implement. I often spend 25minutes+ queuing stuff to run in batch and often something will choke due to a memory spike (IE: I didn't enable VAE tiling). Then I'm stuck. Can't "restart" comfy w/o completely clearing everything I spent time on queuing.
> 
> Please find a way to make this happen (this is a 2023 thread).

The queue and and GUI are horrible tried it too to no avail. The thing is buggy if you build large queues to. I got around it by automating randomised processes and adding extensions to operate the queue.

Don't know if it would still work ATM.

**Comment by ThatCoffeeGuy** [comment]
the fact that this is still not a feature is insane. This extremely unstable app keeps crashing about a few hours, making me lose work, queues, batches.
